cdap-spark-core/src/main/scala/co/cask/cdap/app/runtime/spark/SparkContextCache.scala

```diff
@@ -167,13 +167,9 @@ object SparkContextCache {
     // has fixed this Thread resource leakage or not.
     closers += sc.callMethod("env").orElse(sc.getField("env")).flatMap(env =>
       env.getField("rpcEnv").flatMap(rpcEnv =>
-        rpcEnv.callMethod("fileServer").flatMap(_ match {
-          case s if "org.apache.spark.rpc.netty.HttpBasedFileServer" == s.getClass.getName =>
-            s.getField("httpFileServer")
-          case s if "org.apache.spark.HttpFileServer" == s.getClass.getName => Some(s)
-          case _ => None
-        })
-        .flatMap(fs =>
+        // The rpcEnv.fileServer() call returns either a HttpBasedFileServer, NettyStreamManager or AkkaFileServer
+        // For HttpBasedFileServer or AkkaFileServer, we are interested in the "httpFileServer" field inside
+        rpcEnv.callMethod("fileServer").flatMap(_.getField("httpFileServer")).flatMap(fs =>
           fs.getField("httpServer").flatMap(httpServer => createServerCloser(httpServer.getField("server"), sc))
```

Comment:
All these are wrapped with `Option`. So it there is any field/method missing or exception in call them, they will just get ignored.